au.org.access-nri/model/spack/environment/deployment/1-0-0.json

```diff
@@ -0,0 +1,307 @@
+                    "type": "array",
+                    "default": [],
+                    "items": {
+                      "type": "string"
+                    }
+                  },
+                  "compiler": {
+                    "type": "array",
+                    "default": [],
+                    "items": {
+                      "type": "string"
+                    }
+                  },
+                  "buildable": {
+                    "type": "boolean",
+                    "default": true
+                  },
+                  "permissions": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                      "read": {
+                        "type": "string",
+                        "enum": [
+                          "user",
+                          "group",
+                          "world"
+                        ]
+                      },
+                      "write": {
+                        "type": "string",
+                        "enum": [
+                          "user",
+                          "group",
+                          "world"
+                        ]
+                      },
+                      "group": {
+                        "type": "string"
+                      }
+                    }
+                  },
+                  "package_attributes": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "patternProperties": {
+                      "\\w+": {}
+                    }
+                  },
+                  "providers": {
+                    "type": "object",
+                    "default": {},
+                    "additionalProperties": false,
+                    "patternProperties": {
+                      "\\w[\\w-]*": {
+                        "type": "array",
+                        "default": [],
+                        "items": {
+                          "type": "string"
+                        }
+                      }
+                    }
+                  },
+                  "variants": {
+                    "oneOf": [
+                      {
+                        "type": "string"
+                      },
+                      {
+                        "type": "array",
+                        "items": {
+                          "type": "string"
+                        }
+                      }
+                    ]
+                  }
+                }
+              }
+            },
+            "patternProperties": {
+              "(?!^all$)(^\\w[\\w-]*)": {
+                "type": "object",
+                "default": {},
+                "additionalProperties": false,
+                "properties": {
+                  "require": {
+                    "type": "array",
+                    "minItems": 1,
+                    "additionalItems": true,
```

Comment:
We are allowing more than one requirement?

Reply:
Yep, in the case where we require certain variants as well as the version string, eg https://github.com/ACCESS-NRI/ACCESS-OM3/pull/5/files#diff-e8582e74fa156f4e5729a850e52b24f2fde2d815c2c9c360f88c4cf90db851abR18-R21

Reply:
And below we're making sure the first element is a spack git reference?

Reply:
Yes the first element, but it can just be a regular version reference, as well. For example we can have something like `spack.packages.openmpi.require:@4.6.1`, which doesn't use `@git` style referencing